torch/lib/c10d/NCCLUtils.hpp

```diff
@@ -92,13 +93,15 @@ class NCCLComm {
 
   // Move assignable
   NCCLComm& operator=(NCCLComm&& other) {
```

Comment:
Don't we need a lock in the move constructor above? And also in the destructor?

Reply:
Move assignment means we'd need to lock both instances, not just one of them.

I think we need to take a look first where this is moved to begin with, before adding locks everywhere. Also, is move assignment even required for this class? Things would be a lot easier if we could simply remove the move assignment operator.

Reply:
remove move assignment sounds good to me. How do you think? @pritamdamania87

Reply:
My feeling is that we should keep the two locks and the move assignment operator since these move constructors and assignment operators were probably added following the [Rule of five](https://en.wikipedia.org/wiki/Rule_of_three_(C%2B%2B_programming)#Rule_of_Five).

Reply:
I'm removing this move assignment, looks like that simply adding two locks can still cause deadlock unless we lock the two locks as well based on this discussion https://stackoverflow.com/questions/29986208/how-should-i-deal-with-mutexes-in-movable-types-in-c. 

since there is no testing or is not trivial to add testing for this function, and I searched there is no code depending on this, I'm removing it